apollo-router/src/cache/redis.rs

```diff
@@ -25,9 +25,14 @@ pub(crate) struct RedisValue<V>(pub(crate) V)
 where
     V: ValueType;
 
+enum RedisConnection {
+    Single(redis::aio::Connection),
+    Cluster(Connection),
+}
+
 #[derive(Clone)]
 pub(crate) struct RedisCacheStorage {
-    inner: Arc<Mutex<Connection>>,
+    inner: Arc<Mutex<RedisConnection>>,
```

Comment:
Use trait object instead with https://docs.rs/redis/latest/redis/trait.AsyncCommands.html

Reply:
Ok I tried with this version, but the code looks cleaner with this `RedisConnection` enum